src/prog_algs/predictors/prediction.py

```diff
@@ -19,11 +19,11 @@ class Prediction():
             Data points for each time in times 
     """
 
-    def __init__(self, times, data):
+    def __init__(self, times : list, data : UncertainData):
```

Comment:
```suggestion
    def __init__(self, times : list, data : list):
```

also, could you update the doc comments to list[] instead of array()